pep-0675.rst

```diff
@@ -881,81 +925,81 @@ illustrates a simple denial of service scenario:
 annotations, then violations will go uncaught.
 
-2. ``cast(Literal[str], non_literal_str)`` could be used to lie to the
-type checker and allow a dynamic string value to masquerade as a
-``Literal[str]``. The same goes for a variable that has type ``Any``.
+2. ``cast(LiteralString, non_literal_string)`` could be used to lie to
+the type checker and allow a dynamic string value to masquerade as a
+``LiteralString``. The same goes for a variable that has type ``Any``.
 
 3. Comments such as ``# type: ignore`` could be used to ignore
 warnings about non-literal strings.
 
 4. Trivial functions could be constructed to convert a ``str`` to a
-``Literal[str]``:
+``LiteralString``:
 
 ::
 
-    def make_literal(s: str) -> Literal[str]:
-        letters: Dict[str, Literal[str]] = {
+    def make_literal(s: str) -> LiteralString:
+        letters: Dict[str, LiteralString] = {
             "A": "A",
             "B": "B",
             ...
         }
-        output: List[Literal[str]] = [letters[c] for c in s]
+        output: List[LiteralString] = [letters[c] for c in s]
         return "".join(output)
 
 
 We could mitigate the above using linting, code review, etc., but
 ultimately a clever, malicious developer attempting to circumvent the
-protections offered by ``Literal[str]`` will always succeed. The
-important thing to remember is that ``Literal[str]`` is not intended
+protections offered by ``LiteralString`` will always succeed. The
+important thing to remember is that ``LiteralString`` is not intended
 to protect against *malicious* developers; it is meant to protect
 against benign developers accidentally using sensitive APIs in a
 dangerous way (without getting in their way otherwise).
 
-Without ``Literal[str]``, the best enforcement tool API authors have
+Without ``LiteralString``, the best enforcement tool API authors have
 is documentation, which is easily ignored and often not seen. With
-``Literal[str]``, API misuse requires conscious thought and artifacts
+``LiteralString``, API misuse requires conscious thought and artifacts
 in the code that reviewers and future developers can notice.
 
 .. _appendix_C:
 
-Appendix C: ``str`` methods that preserve ``Literal[str]``
-==========================================================
+Appendix C: ``str`` methods that preserve ``LiteralString``
+===========================================================
 
 The ``str`` class has several methods that would benefit from
-``Literal[str]``. For example, users might expect
-``"hello".capitalize()`` to have the type ``Literal[str]`` similar to
-the other examples we have seen in the `Inferring Literal[str]
-<inferring_literal_str>`_ section. Inferring the type ``Literal[str]``
+``LiteralString``. For example, users might expect
+``"hello".capitalize()`` to have the type ``LiteralString`` similar to
+the other examples we have seen in the `Inferring LiteralString
+<inferring_literal_string>`_ section. Inferring the type ``LiteralString``
```

Comment:
```suggestion
<inferring_literal_string_>`_ section. Inferring the type ``LiteralString``
```

Syntax mistake

Reply:
I swear RST is the least memorizable syntax I've ever come across. I can never recall how to link to something :| Thank goodness we have Markdown elsewhere.

(Why didn't the RST pre-commit checks catch this?)

Reply:
> I swear RST is the least memorizable syntax I've ever come across. I can never recall how to link to something :| Thank goodness we have Markdown elsewhere.

Heh, no worries; I've had multiple PEP editors (including the writer of the new build system) think this wasn't required, due to the different rules for implicit references vs. explicit reference roles. There are a number of things that could be more intuitive and user-friendly about RST, links being the worst offender, but with Sphinx it is far more powerful than Markdown when it comes to its functionality for documentation and things like the PEP repo; a lot of what we do here simply wouldn't be possible without RST's extensible roles, directives and structured style.

> (Why didn't the RST pre-commit checks catch this?)

There isn't currently a check for this, but I could write a new one. However, assuming PEP-0676 is accepted, once the new Sphinx build system is the default, you can just use the standard `:ref:` role to link sections, which doesn't have this problem and also has the section name as its default inline text, so instead of

```rst
`Inferring LiteralString <inferring_literal_string_>`_
```

you could just do

```rst
:ref:`inferring_literal_string`
```